src/components/Dialog/Dialog.tsx

```diff
@@ -31,6 +31,8 @@ export interface DialogProps extends TPAComponentProps {
   closeButtonAriaLabel?: string;
   /** Identifies the element that labels the close button element. Optional. */
   closeButtonAriaLabelledby?: string;
+  /** Whether the Dialog is wired to the site palette, or has a white background */
+  wiredToSiteColors?: boolean;
```

Comment:
following our discussion. since the component knows it should not be wired. I think we can keep this logic internal.

wrap the children with

```
<TPAComponentsProvider value ={{mobile, rtl, wiredToSiteColors}}>
...
</TPAComponentsProvider>
```

and then, every component (`IconButton` in this case) will apply a `black-and-white` skin

In the rare case that some would like to opt-out the black and white theme, only then he should pass `wiredToSiteColors=true`

Reply:
So I actually thought that we'll have a different provider for this, and the Dialog component will use it.
Since the `TPAComponentsProvider` is in the app level, I'm not sure it's the right place.

So each component will use this context:
```tsx
render() {
  ...
  return (
    <BnWContext.Consumer>
      {
        (isBnW) => (
          ...
        )
      }
    </BnWContext.Consumer>
  )
}
```

and the `Dialog` component will use the provider:
```tsx
render() {
  ...
  return (
    <BnWContext.Provider value={{ isBnW: true }}>
      ...
    </BnWContext.Provider>
  );
}
```

Reply:
and also, I preferred working on this in another PR

Reply:
Do you think the provider will have more options than just `isBnW`?
Not sure the use-case justifies a completely new provider at the moment

Reply:
We have to use a new provider, since our `TPAComponentsProvider` is used in the application level, passing application level properties.
In this case we need a specific context for components inside the Dialog (or inside this new provider, for that matter).
One provider won't work.
In any case, this will be the next PR